modules/internal/ChapelBase.chpl

```diff
@@ -840,8 +841,14 @@ module ChapelBase {
   // locale as the sync/cofall/cobegin was initiated on and thus the
   // same locale on which the object is allocated.
   pragma "dont disable remote value forwarding"
-  inline proc _endCountAlloc() {
-    return new _EndCount();
+  inline proc _endCountAlloc(param forceLocalTypes : bool) {
```

Comment:
Don't you want two params here? One for forcing local types of `i` and one for forcing it for `taskCnt`

I think the only time we want a network atomic for `taskCnt` is for remote begin statements, but right now I think we'll use them for `coforall on's` too.